st2api/st2api/controllers/v1/actionexecutions.py

```diff
@@ -129,11 +146,14 @@ def _get_children(self, id_, depth=-1, result_fmt=None):
         # make sure depth is int. Url encoding will make it a string and needs to
         # be converted back in that case.
         depth = int(depth)
+        from_model_kwargs = self._get_from_model_kwargs_for_request(request=pecan.request)
         LOG.debug('retrieving children for id: %s with depth: %s', id_, depth)
         descendants = execution_service.get_descendants(actionexecution_id=id_,
                                                         descendant_depth=depth,
                                                         result_fmt=result_fmt)
-        return [self.model.from_model(descendant) for descendant in descendants]
```

Comment:
It can, a lot of methods we call in the API handlers can (and sometimes do throw) and we do nothing about it.

I'm not saying that's good, I'm just saying we first need to decide on a consistent way of handling exception in the API.